### PR TITLE
ci: fix chocolatey manifest

### PR DESCRIPTION
**Pull request by phorcys420**
[image](https://community.chocolatey.org/packages/coder/2.1.1)

---

Now that the package does build, there are still some issues and the package is getting rejected automatically.

- [X] The IconUrl element in the nuspec file should be a valid Url. Please correct this [More...](https://docs.chocolatey.org/en-us/community-repository/moderation/package-validator/rules/cpmr0031)
- [x] Binary files (.exe, .msi, .zip, etc) have been included without including a LICENSE.txt file. This file is required when including binaries [More...](https://docs.chocolatey.org/en-us/community-repository/moderation/package-validator/rules/cpmr0005)
- [x] Binary files (.exe, .msi, .zip) have been included without including a VERIFICATION.txt file. This file is required when including binaries [More...](https://docs.chocolatey.org/en-us/community-repository/moderation/package-validator/rules/cpmr0006)
- [x] Release Notes (releaseNotes) are a short description of changes in each version of a package. Please include releasenotes in the nuspec. NOTE: To prevent the need to continually update this field, providing a URL to an external list of Release Notes is perfectly acceptable.

I need to look into VERIFICATION.txt, but I'm pretty sure that it's simply a list of checksums or something similar.

---

On the other hand, we have the LICENSE problem again.
I did not know this but Chocolatey forces you to provide both a LICENSE.txt and a licenseUrl.

The options we have are:
1. Fetching the terms of service and adding them to a file (I don't like this because the terms of service weren't really a license in the first place)
2. Crafting a dual-license and exposing it somehow via the web (see LICENSE.mixed ideas from 
3. Rewriting the Chocolatey package to download directly from the GH source (this skips the need for LICENSE.txt and also allows more architectures than only x64)

I think the last two are the best options.

---

(cc: @bpmct)

### CODE REVIEW

**Comment by deansheather** [comment]
We definitely need `LICENSE.mixed` or something in the root of the repo. The windows installer script's generated license can be used as the basis for this, and the script can be updated to just use it directly.

Regarding verification file, just a simple note saying the following should suffice:

```
SHA256 checksums are available on the official GitHub releases page: <LINK>

The binary shipped via Chocolatey is `coder_${VERSION}_windows_amd64.exe`. The binary can be verified with `Get-FileHash -Algorithm SHA256 binary.exe` and comparing it to the checksum in the checksums file.
```

File: scripts/chocolatey/coder.nuspec [hunk 1]
```diff
@@ -22,13 +22,14 @@
     <title>Coder (Install)</title>
```
Comment (matifali) [inline scripts/chocolatey/coder.nuspec:22]:
```suggestion
    <title>Coder</title>
```
Why do we have (Install) as suffix here?

Comment (phorcys420) [reply scripts/chocolatey/coder.nuspec]: It's a pretty standard thing, but we don't need it, I did it on accident

File: scripts/chocolatey/coder.nuspec [hunk 2]
```diff
@@ -22,13 +22,14 @@
...
     <projectUrl>https://coder.com</projectUrl>
-    <iconUrl>https://github.com/coder/presskit/raw/main/logos/coder%20logo%20black%20square.png?raw=true</iconUrl>
+    <iconUrl>https://github.com/coder/presskit/blob/main/logos/coder_logo_black_square.png?raw=true</iconUrl>
     <copyright>Coder Technologies, Inc.</copyright>
     <licenseUrl>https://coder.com/legal/terms-of-service</licenseUrl>
     <requireLicenseAcceptance>true</requireLicenseAcceptance>
     <projectSourceUrl>https://github.com/coder/coder.git</projectSourceUrl>
     <docsUrl>https://coder.com/docs/v2/latest</docsUrl>
     <bugTrackerUrl>https://github.com/coder/coder/issues</bugTrackerUrl>
+		<releaseNotes>https://github.com/coder/coder/releases/tag/v$version$</releaseNotes>
```
[the remark this answers is not in the document]
Comment (phorcys420) [reply scripts/chocolatey/coder.nuspec]: yes, it does, I checked

File: scripts/chocolatey/coder.nuspec [hunk 3]
```diff
@@ -22,13 +22,14 @@
...
     <tags>coder remote-dev terraform development</tags>
```
[the remark this answers is not in the document]
Comment (phorcys420) [reply scripts/chocolatey/coder.nuspec:33]: I don't see why not

File: scripts/chocolatey/coder.nuspec (hunk 1 above)
Comment (phorcys420) [reply scripts/chocolatey/coder.nuspec]: removed in f7d2695

File: scripts/chocolatey/coder.nuspec (hunk 3 above)
[the remark this answers is not in the document]
Comment (phorcys420) [reply scripts/chocolatey/coder.nuspec]: done in b9cee27

**Comment by phorcys420** [comment]
currently, I'm going to create a VERIFICATION file, but the long term goal will be to remove it and move to powershell rather than packing binaries.

VERIFICATION is needed only when packing binaries, also moving to powershell will allow for automatic checksum verification.
it will also help for different architectures as I said before.

I will look into it in the future but for now it's faster to create a VERIFICATION file.

**Comment by phorcys420** [comment]
> SHA256 checksums are available on the official GitHub releases page: <LINK>
> 
> The binary shipped via Chocolatey is `coder_${VERSION}_windows_amd64.exe`. The binary can be verified with `Get-FileHash -Algorithm SHA256 binary.exe` and comparing it to the checksum in the checksums file.

I think we should include the checksum in the file, I am not sure that the chocolatey mods will like having to visit a separate link for this

**Comment by phorcys420** [comment]
@deansheather The example Chocolatey VERIFICATION file says "<If software vendor, explain that here - checksum verification instructions are optional>"

```
Note: Include this file if including binaries you have the right to distribute.
Otherwise delete. this file. If you are the software author, you can change this
mention you are the author of the software.

===DELETE ABOVE THIS LINE AND THIS LINE===

VERIFICATION
Verification is intended to assist the Chocolatey moderators and community
in verifying that this package's contents are trustworthy.

<Include details of how to verify checksum contents>
<If software vendor, explain that here - checksum verification instructions are optional>
```

Therefore, I suggest a ``VERIFICATION`` file that looks like this, what do you think ?
```md
VERIFICATION
Verification is intended to assist the Chocolatey moderators and community
in verifying that this package's contents are trustworthy.

This packaged is [published](https://github.com/coder/coder/blob/main/.github/workflows/release.yaml) automatically by the software vendor/author using GitHub Actions.
```

**Comment by phorcys420** [comment]
this should now be "good to merge", but I will leave this as a draft since the license thing isn't solved yet.
I pre-emptively changed the license to ``LICENSE.enterprise`` because kyle suggested that it will be modified to extend the AGPL (see discussion in #9269)

**Comment by deansheather** [comment]
```
VERIFICATION
Verification is intended to assist the Chocolatey moderators and community
in verifying that this package's contents are trustworthy.

This package is automatically published by Coder Technologies, Inc., the vendor of this software.

See the https://github.com/coder/coder release workflow.
```

should be good enough

**Comment by phorcys420** [comment]
Done, I took the liberty of using Markdown links, since Chocolatey also accepts ``md`` files for ``VERIFICATION`` and ``LICENSE``.

**Comment by matifali** [comment]
@deansheather, @kylecarbs, we are currently not publishing a chocolatey package.

**Comment by deansheather** [comment]
TBH we already publish winget which is installed by default on windows, so I don't see the value in chocolatey (also the packaging process seems really tedious)

**Comment by matifali** [comment]
@deansheather ok, I agree with you on this. We can close it and remove the publish choco job from `ci.yaml`.

**Comment by matifali** [comment]
@phorcys420 Considering the license issue, and we already publish on `winget`, we have decided not to publish with a chocolatey package. I will remove it from CI in a follow-up PR. Thanks a lot for doing this.

**Pull request closed without merging**